x/pricefeed/keeper/keeper.go

```diff
@@ -102,11 +113,21 @@ func (k Keeper) SetCurrentPrices(ctx sdk.Context, marketID string) sdk.Error {
 	}
 	medianPrice := k.CalculateMedianPrice(ctx, notExpiredPrices)
 
+	// Emit an event for market price update
+	ctx.EventManager().EmitEvent(
+		sdk.NewEvent(
+			types.EventTypeMarketPriceUpdated,
+			sdk.NewAttribute(types.AttributeMarketID, fmt.Sprintf("%s", marketID)),
+			sdk.NewAttribute(types.AttributeMarketPrice, fmt.Sprintf("%s", medianPrice.String())),
+		),
+	)
+
```

Comment:
nit: this will ouput an event every block even if the price hasn't changed

Reply:
One option is to remove this event and replace it with one that only emits if the price changes. This would requires storing the previous value of each market. I don't have a strong preference between keeping it as-is or changing it.

Reply:
I did a thing in https://github.com/Kava-Labs/kava/pull/343/commits/fad21d56f2ccf03a6a75cd299991af0263397af1 but can't test it since it seems only events emitted by txs can be queried. It doesn't break anything but idk if it's working properly.